File: backend/src/main/java/ch/helsanahack/backend/domain/UserData.java
package ch.helsanahack.backend.domain;

import ch.helsanahack.backend.entities.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserData {
    private UserId userId;
    private BmiEntry bmiEntry;
    private List<CalorieBurnedEntry> calorieBurnedEntries;
    private List<PulseEntry> pulseEntries;
    private List<SleepEntry> sleepEntries;
    private List<StepEntry> stepEntries;
}

File: frontend/src/models/UserData.ts
import {CalorieBurnedEntry} from "src/models/CalorieBurnedEntry";
import {PulseEntry} from "src/models/PulseEntry";
import {SleepEntry} from "src/models/SleepEntry";
import {StepEntry} from "src/models/StepEntry";
import {BmiEntry} from "src/models/BmiEntry";

export class UserData {
  userId: string;
  bmiEntry: BmiEntry;
  calorieBurnedEntries: CalorieBurnedEntry[];
  pulseEntries: PulseEntry[];
  sleepEntries: SleepEntry[];
  stepEntries: StepEntry[];

  constructor(userId: string, bmiEntry: BmiEntry, calorieBurnedEntries: CalorieBurnedEntry[], pulseEntries: PulseEntry[],
              sleepEntries: SleepEntry[], stepEntries: StepEntry[]) {
    this.userId = userId;
    this.bmiEntry = bmiEntry;
    this.calorieBurnedEntries = calorieBurnedEntries;
    this.pulseEntries = pulseEntries;
    this.sleepEntries = sleepEntries;
    this.stepEntries = stepEntries;
  }
}
